lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1140,6 +1139,12 @@ WasmBytecodeGenerator::EmitMemAccess(WasmOp wasmOp, const WasmTypes::WasmType* s
     {
         throw WasmCompilationException(_u("Index expression must be of type I32"));
     }
+
+    if (offset >= m_module->GetMemory()->maxSize)
+    {
+        throw WasmCompilationException(_u("Index is out of bounds"));
```

Comment:
No, we can't know this at compile time. Offset+index might overflow where offset alone doesn't. Besides this doesn't work for import or grow memory